azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/service/sdk/ChatEventHandler.kt

```diff
@@ -0,0 +1,249 @@
+    private val participantAddedEvent =
+        ChatEventWrapper(ChatEventType.PARTICIPANTS_ADDED, eventReceiver)
+    private val participantRemovedEvent =
+        ChatEventWrapper(ChatEventType.PARTICIPANTS_REMOVED, eventReceiver)
+
+    private lateinit var chatThreadID: String
+    private lateinit var eventSubscriber: (ChatEventModel) -> Unit
+
+    fun start(
+        chatClient: ChatClient,
+        threadID: String,
+        eventSubscriber: (ChatEventModel) -> Unit
+    ) {
+
+        this.chatThreadID = threadID
+        this.eventSubscriber = eventSubscriber
+
+        chatClient.addEventHandler(ChatEventType.CHAT_MESSAGE_RECEIVED, messageReceivedEvent)
+        chatClient.addEventHandler(ChatEventType.CHAT_MESSAGE_EDITED, messageEditedEvent)
+        chatClient.addEventHandler(ChatEventType.CHAT_MESSAGE_DELETED, messageDeletedEvent)
+        chatClient.addEventHandler(
+            ChatEventType.TYPING_INDICATOR_RECEIVED,
+            typingIndicatorReceivedEvent
+        )
+        chatClient.addEventHandler(ChatEventType.READ_RECEIPT_RECEIVED, readReceiptReceivedEvent)
+        chatClient.addEventHandler(ChatEventType.CHAT_THREAD_CREATED, chatThreadCreatedEvent)
+        chatClient.addEventHandler(ChatEventType.CHAT_THREAD_DELETED, chatThreadDeletedEvent)
+        chatClient.addEventHandler(
+            ChatEventType.CHAT_THREAD_PROPERTIES_UPDATED,
+            chatThreadPropertiesUpdatedEvent
+        )
+        chatClient.addEventHandler(ChatEventType.PARTICIPANTS_ADDED, participantAddedEvent)
+        chatClient.addEventHandler(ChatEventType.PARTICIPANTS_REMOVED, participantRemovedEvent)
+    }
+
+    fun stop(chatClient: ChatClient) {
+        chatClient.removeEventHandler(ChatEventType.CHAT_MESSAGE_RECEIVED, messageReceivedEvent)
+        chatClient.removeEventHandler(ChatEventType.CHAT_MESSAGE_EDITED, messageEditedEvent)
+        chatClient.removeEventHandler(ChatEventType.CHAT_MESSAGE_DELETED, messageDeletedEvent)
+        chatClient.removeEventHandler(
+            ChatEventType.TYPING_INDICATOR_RECEIVED,
+            typingIndicatorReceivedEvent
+        )
+        chatClient.removeEventHandler(ChatEventType.READ_RECEIPT_RECEIVED, readReceiptReceivedEvent)
+        chatClient.removeEventHandler(ChatEventType.CHAT_THREAD_CREATED, chatThreadCreatedEvent)
+        chatClient.removeEventHandler(ChatEventType.CHAT_THREAD_DELETED, chatThreadDeletedEvent)
+        chatClient.removeEventHandler(
+            ChatEventType.CHAT_THREAD_PROPERTIES_UPDATED,
+            chatThreadPropertiesUpdatedEvent
+        )
+        chatClient.removeEventHandler(ChatEventType.PARTICIPANTS_ADDED, participantAddedEvent)
+        chatClient.removeEventHandler(ChatEventType.PARTICIPANTS_REMOVED, participantRemovedEvent)
+    }
+
+    private fun onEventReceived(eventType: ChatEventType, chatEvent: ChatEvent) {
+        if (chatThreadID != chatEvent.chatThreadId) {
+            return
+        }
+
+        when (eventType) {
```

Comment:
This method is too long, is it possible to refactor into functions? Same as before, is it possible to make casting safe? E.g. when(chatEvent) instead of when (eventType).

If Typing on ChatEvent, is ChatEvenType necessary?

Reply:
Event type is from native SDK. It is possible that in future two events can have same EventObjectWith data. 
We can create functions to split. However, specific class is created only for handling events.